frontend/src/pages/AccountSend.vue

```diff
@@ -603,12 +603,12 @@ function useSendForm() {
     let ethTotal;
     for (const token of summaryAmount.value.keys()) {
       if (token.symbol === NATIVE_TOKEN.value.symbol) {
-        ethTotal = summaryAmount.value.get(token);
+        ethTotal = summaryAmount.value.get(token)?.toString();
       }
     }
     if (typeof ethTotal !== 'string') return '--'; // appease TS
     if (isNaN(Number(ethTotal))) return '--';
-    const sendAmount = parseUnits(ethTotal, NATIVE_TOKEN.value.decimals);
+    const sendAmount = BigNumber.from(ethTotal);
     const totalAmount = sendAmount.add(toll.value.mul(batchSends.value.length));
 
     return humanizeArithmeticResult(
```

Comment:
I think we can clean this up a bit now:
- On L606, remove the `.toString()` so the type of `ethTotal` is either `BigNumber | undefined`
- L609-610 can then be collapsed into `if (!ethTotal) return '--';
- L611 can be removed now because `ethTotal` is already a `BigNumber`

I haven't tested this code but it should be the same
```typescript
const batchSendHumanTotalAmount = computed(() => {
  let ethSendAmount: BigNumber | undefined;
  for (const token of summaryAmount.value.keys()) {
    if (token.symbol === NATIVE_TOKEN.value.symbol) {
      ethSendAmount = summaryAmount.value.get(token);
    }
  }
  if (!ethSendAmount) return '--';
  const totalAmount = ethSendAmount.add(toll.value.mul(batchSends.value.length));

  return humanizeArithmeticResult(
    totalAmount,
    [ethSendAmount.toString(), batchSendHumanToll.value], // subtotal and fee
    NATIVE_TOKEN.value
  );
});
```

Reply:
Updated here: https://github.com/ScopeLift/umbra-protocol/pull/608/commits/3182021b042ae7e432d87cb848fffebf018f00f0